### PR TITLE
Test layout content changes

### PR DESCRIPTION
**Pull request by swankjesse**
We were missing tests for items being removed from Column, etc.

### CODE REVIEW

File: redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt [hunk 1]
```diff
@@ -429,6 +451,8 @@ interface TestFlexContainer<T : Any> : Widget<T>, ChangeListener {
   fun mainAxisAlignment(mainAxisAlignment: MainAxisAlignment)
   fun margin(margin: Margin)
   fun add(widget: Widget<T>)
+  fun addAt(index: Int, widget: Widget<T>)
```
Comment (swankjesse) [inline redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt:454]: The first thing I tried was exposing the `Widget.Children` list in this facet. That doesn’t work so well 'cause LazyList doesn’t know its own count until `onEndChanges()` is called.

File: redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiFlexContainerTest_testDynamicElementUpdates[LTR]_abcde.png [hunk 2]
```diff
@@ -0,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
```
Comment (JakeWharton) [inline redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiFlexContainerTest_testDynamicElementUpdates[LTR]_abcde.png:1]: Why do these have lowercase suffixes? Did you start with lowercase and then switch to uppercase on a case-insensitive file system? The iOS ones have uppercase suffixes like I would expect.

Makes me wonder if we can force CI onto a case sensitive FS on Mac...

**Review by JakeWharton — APPROVED** [review]
Can't see the images on mobile app, but I'll trust they're right.

File: redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiFlexContainerTest_testDynamicElementUpdates[LTR]_abcde.png (hunk 2 above)
Comment (swankjesse) [reply redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiFlexContainerTest_testDynamicElementUpdates[LTR]_abcde.png:1]: Good catch, yeah that’s super weird. I started uppercase and I’m using a case-sensitive file system. I wonder if Paparazzi does this?!

Comment (swankjesse) [reply redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiFlexContainerTest_testDynamicElementUpdates[LTR]_abcde.png:1]: Confirmed.
https://github.com/cashapp/paparazzi/blob/master/paparazzi/src/main/java/app/cash/paparazzi/Snapshot.kt#L36

**Pull request merged**